### PR TITLE
Face mask

### LINKED ISSUE #510: Mask or not
I would like to add a project for face mask detection using CNN. Here, I would use dataset which contains real life images of human faces.

### PR DESCRIPTION
**Pull request by sudipg4112001**
### Related Issue
Fixes #510

### CODE REVIEW

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Just a few modifications before merging:
- Add a section in readme named "Setup Instructions" where describe that extract the dataset.rar  and how to run the script.
- Update all the paths in script as shown below in example:
```python
testing_target=data_set('./Face-Mask-Detection/kaggle/input/face-mask-detection/test/')
```
-  Shift all the relevant import statements on top

**Comment by sudipg4112001** [comment]
I have made the necessary changes. Please review it.

File: Face-Mask-Detection/face_mask_detection_using_cnn.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+from tensorflow.keras import layers
+import tensorflow as tf
+from keras.preprocessing.image import img_to_array
+from tensorflow.keras import backend
+from tensorflow.keras.preprocessing.image import ImageDataGenerator
+import os
+from sklearn.metrics import classification_report
+import sklearn.metrics as metrics
+import itertools
+for dirname, _, filenames in os.walk('/kaggle/input'):
```
Comment (kaustubhgupta) [inline Face-Mask-Detection/face_mask_detection_using_cnn.py:15]: This path also needs modification

File: Face-Mask-Detection/face_mask_detection_using_cnn.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                continue
+        if(df1['class'][i]=='mask'):
+            without_mask='with_mask'
+        image=img_to_array(image)
+        data.append(image)
+        target.append(data_map[without_mask])
+    data = np.array(data, dtype="float") / 255.0
+    target = tf.keras.utils.to_categorical(np.array(target), num_classes=2)
+    return data, target
+training_target=data_set('./Face-Mask-Detection/kaggle/input/face-mask-detection/train/')
```
Comment (kaustubhgupta) [inline Face-Mask-Detection/face_mask_detection_using_cnn.py:54]: What do you want to load here? There is no extension to files. Same for next testing_target and valid_target. Also, where are the testing_data, training_data, and valid_data variables? They are not initialized anywhere but passed into functions.

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Run the whole script on your system and debug the errors. Push your changes once you are confirmed that the script runs and there is no error from your side. Found some while reviewing only:

File: Face-Mask-Detection/face_mask_detection_using_cnn.py (hunk 2 above)
Comment (antrikshmisri) [reply Face-Mask-Detection/face_mask_detection_using_cnn.py:54]: @kaustubhgupta I think he is specifying the file [here](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/a6c083ed747952a453825cbdfa022522eb1bcc6e/Face-Mask-Detection/face_mask_detection_using_cnn.py#L23-L25) , I could be wrong as I haven't run this code yet

Comment (kaustubhgupta) [reply Face-Mask-Detection/face_mask_detection_using_cnn.py:54]: Actually the pylint test gave these errors and when I saw the script, I found these variables are missing but passed in functions

Comment (antrikshmisri) [reply Face-Mask-Detection/face_mask_detection_using_cnn.py:54]: @kaustubhgupta Yup , the linter is giving error because of undefined variables. I am referring to the training_target variable , he is taking directory as argument to data_set function and then appending file name [here](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/a6c083ed747952a453825cbdfa022522eb1bcc6e/Face-Mask-Detection/face_mask_detection_using_cnn.py#L23-L25).

**Comment by sudipg4112001** [comment]
What should I do ?

**Comment by kaustubhgupta** [comment]
> What should I do ?

all the suggestions are mentioned above

**Comment by sudipg4112001** [comment]
I have done the changes, please do review

File: Face-Mask-Detection/face_mask_detection_using_cnn.py (hunk 2 above)
Comment (sudipg4112001) [reply Face-Mask-Detection/face_mask_detection_using_cnn.py]: > What do you want to load here? There is no extension to files. Same for next testing_target and valid_target. Also, where are the testing_data, training_data, and valid_data variables? They are not initialized anywhere but passed into functions.

extension is not required, because they are folders and they will directly have access to the files within them.

**Comment by kaustubhgupta** [comment]
@antrikshmisri I ran the script but it didn't work. What should we do?

**Comment by antrikshmisri** [comment]
> @antrikshmisri I ran the script but it didn't work. What should we do?

@kaustubhgupta I am unable to get some dependencies so , couldn't test it. @santushtisharma10 CAn you confirm the same

**Comment by sudipg4112001** [comment]
I need to do something ?

**Comment by santushtisharma10** [comment]
@antrikshmisri @kaustubhgupta facing following error while running the script
image

**Comment by kaustubhgupta** [comment]
> @antrikshmisri @kaustubhgupta facing following error while running the script
> image

@sudipg4112001 Can you guide how to fix this? Is the script running on your machine? Can you share any GIF, video as Proof?

**Comment by sudipg4112001** [comment]
Yes, this same code is running smoothly in my device. I am confused what's the problem with yours. I am attaching a screenshot below.

Face-Mask-Detection-ss

**Comment by kaustubhgupta** [comment]
> Yes, this same code is running smoothly in my device. I am confused what's the problem with yours. I am attaching a screenshot below.
> 
> Face-Mask-Detection-ss
> 
> After making the changes you suggested it's now running smoothly.

As he has shown this screenshot, should we approve this? @antrikshmisri @santushtisharma10

**Comment by antrikshmisri** [comment]
> > Yes, this same code is running smoothly in my device. I am confused what's the problem with yours. I am attaching a screenshot below.
> > Face-Mask-Detection-ss
> > After making the changes you suggested it's now running smoothly.
> 
> As he has shown this screenshot, should we approve this? @antrikshmisri @santushtisharma10

@kaustubhgupta @santushtisharma10 Okay from my side but I think we should refer @avinashkranjan first.

**Comment by kaustubhgupta** [comment]
> > > Yes, this same code is running smoothly in my device. I am confused what's the problem with yours. I am attaching a screenshot below.
> > > Face-Mask-Detection-ss
> > > After making the changes you suggested it's now running smoothly.
> > 
> > 
> > As he has shown this screenshot, should we approve this? @antrikshmisri @santushtisharma10
> 
> @kaustubhgupta @santushtisharma10 Okay from my side but I think we should refer @avinashkranjan first.

Yes, that makes sense

**Comment by kaustubhgupta** [comment]
Updates: Conveyed the message to PA, he will respond soon

**Comment by avinashkranjan** [comment]
**The Scripts Seems to working for me..**
*_Just Correct This Error_*
Screenshot 2021-03-22 000445

I Just Commented out **line 39**..And the **Script works fine for me** 
Screenshot 2021-03-22 000750

So, I'm approving the Script..Make the required changes in **line 39**, Else Remove it..Will Merge it then

**Comment by sudipg4112001** [comment]
> **The Scripts Seems to working for me..**
> __Just Correct This Error__
> Screenshot 2021-03-22 000445
> 
> I Just Commented out **line 39**..And the **Script works fine for me**
> Screenshot 2021-03-22 000750
> 
> So, I'm approving the Script..Make the required changes in **line 39**, Else Remove it..Will Merge it then

I am commenting out line 39. So that in future if it is required it can be reused.

**Comment by sudipg4112001** [comment]
I have done the required changes...

**Comment by sudipg4112001** [comment]
I have made the changes..

**Comment by sudipg4112001** [comment]
Do review

**Comment by avinashkranjan** [comment]
@sudipg4112001 Don't Spam the Comments..Will Review it in Sometime..

**Review by avinashkranjan — APPROVED** [review]
LGTM..💯🚩
